src/components/KTruncate/KTruncate.vue

```diff
@@ -0,0 +1,280 @@
+    for (let i = 0; i < children.length; i++) {
+      // find height of tallest child
+      tallestChildHeight = children[i].offsetHeight > tallestChildHeight ? children[i].offsetHeight : tallestChildHeight
+    }
+    const targetWrapperHeight = (props.rows === 1 ? 0 : (props.rows - 1) * 10) + (tallestChildHeight * props.rows)
+    wrapperHeight.value = kTruncateContainer.value.offsetHeight > targetWrapperHeight ? `${targetWrapperHeight}px` : 'auto'
+  }
+}
+
+/**
+ * Determines whether expand/collapse toggle need to be shown.
+ * In other words, determines whether the content overflows.
+ * If the height of the content is greater than height of the wrapper element - content does overflow.
+ */
+const updateToggleVisibility = (): void => {
+  setWrapperHeight()
+  if (kTruncateContainer.value && kTruncateWrapper.value) {
+    // in case with text content, need to compare scrollHeight value
+    const containerHeightProperty = props.isTextContent ? kTruncateContainer.value.scrollHeight : kTruncateContainer.value.offsetHeight
+    const textToggleControlsHeight = textToggleControls.value ? textToggleControls.value.offsetHeight : 0
+    /**
+     * In case with text content, toggle controls element is rendered below content, so adds up to wrapper height.
+     * Therefore it's height need to be subtracted to determine the actual wrapper height.
+     */
+    showToggle.value = containerHeightProperty > (kTruncateWrapper.value.offsetHeight - textToggleControlsHeight)
+    countExcessElements()
+  }
+}
+
+// Counts elements that are wrapped to the hidden rows and therefore are not visible.
+const countExcessElements = (): void => {
+  if (props.isTextContent) {
+    return
+  }
+
+  excessElementsCount.value = 0
+  if (kTruncateContainer.value && kTruncateWrapper.value) {
+    const children = kTruncateContainer.value.children as unknown as HTMLElement[]
+    for (let i = 0; i < children.length; i++) {
+      /**
+       * If element's offsetTop
+       * (offset from the nearest relatively positioned parent, which is .k-truncate-container)
+       * is greater than the wrapper element height - means it's not visible
+       */
+      if (children[i].offsetTop > kTruncateWrapper.value.offsetHeight) {
+        excessElementsCount.value += 1
+      }
+    }
+  }
+}
+
+const handleToggleClick = async () => {
+  expanded.value = !expanded.value
+  // await for component to collapse/expand
+  await nextTick()
+  updateToggleVisibility()
+}
+
+onMounted(() => {
+  resizeObserver.value = new ResizeObserver(updateToggleVisibility).observe(kTruncateContainer.value as HTMLDivElement)
+})
+
+onBeforeUnmount(() => {
+  if (resizeObserver.value) {
+    resizeObserver.value.unobserve(kTruncateContainer.value)
+  }
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
+@import '@/styles/functions';
+
+.k-truncate {
+  align-items: flex-start;
+  display: flex;
+  overflow: hidden;
+  width: 100%;
+
+  .expand-trigger, .collapse-trigger {
```

Comment:
nit: move to new line
```suggestion
  .expand-trigger, 
  .collapse-trigger {
```